fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/prunesubfield/PruneSubfieldRule.java

```diff
@@ -48,25 +60,21 @@ public PruneSubfieldRule() {
     public List<OptExpression> transform(OptExpression input, OptimizerContext context) {
         // project expression
         LogicalProjectOperator project = input.getOp().cast();
-        SubfieldExpressionCollector collector = new SubfieldExpressionCollector();
+        SubfieldExpressionCollector collector =
+                new SubfieldExpressionCollector(context.getSessionVariable().isCboPruneJsonSubfield());
         for (ScalarOperator value : project.getColumnRefMap().values()) {
             value.accept(collector, null);
         }
 
-        List<ScalarOperator> allSubfieldExpr = Lists.newArrayList();
-        allSubfieldExpr.addAll(collector.getComplexExpressions());
-
         // scan predicate
         LogicalScanOperator scan = input.getInputs().get(0).getOp().cast();
```

Comment:
another Rule `PushDownSubfieldRule` will push expression to scan. and as you say, the project can't pushed down to the scan in some scenarios.

The previous version of the sub-field prune framework don't depend on `project-scan` mode, but I refactor it because there was a significant risk of crashe